src/main/java/ro/unibuc/hello/exception/BadRequestException.java

```diff
@@ -0,0 +1,12 @@
+package ro.unibuc.hello.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(value = HttpStatus.BAD_REQUEST)
```

Comment:
tldr Defining custom exceptions for each HTTP status does not bring additional value/new information.

I don't think that's quite the intended way to use ResponseStatus. I think you should use it to annotate exceptions related to the business logic (like UserNotFound, InvalidTaskId, etc.). Otherwise, you could use ResponseEntity to set the response status.

Reply:
I added a BadRequestException with the intent of adding later a Global Exception Handler, which will return a ResponseEntity with the exception's message and status code, just my preference

Reply:
Then shouldn't you still throw business logic-related exceptions? They will be handled by a global handler too.

Reply:
Yes but i think it's a little overkill to make a custom exception for every piece of business, as you can easily reach 10+